### PR TITLE
bpo-36057 Update docs and tests for ordering in collections.Counter [no behavior change]

### PR DESCRIPTION
**Pull request by rhettinger**
https://bugs.python.org/issue36057

### CODE REVIEW

File: Doc/library/collections.rst [hunk 1]
```diff
@@ -253,15 +253,20 @@ For example::
 
     .. versionadded:: 3.1
 
+    .. versionchanged:: 3.7 As a :class:`dict` subclass, :class:`Counter`
+       inherited the capability to remember insertion order.  Math operations
```
Comment (eamanu) [inline Doc/library/collections.rst]: Should start with capital? `Inherited`

File: Lib/test/test_collections.py [hunk 2]
```diff
@@ -1867,6 +1867,63 @@ def test_init(self):
...
         self.assertRaises(TypeError, Counter.__init__)
 
+    def test_order_preservation(self):
+        # Input order dictates items() order
+        self.assertEqual(list(Counter('abracadabra').items()),
+               [('a', 5), ('b', 2), ('r', 2), ('c', 1), ('d', 1)])
+        # letters with same count:   ^----------^         ^---------^
+
+        # Verify retention of order even when all counts are equal
+        self.assertEqual(list(Counter('xyzpdqqdpzyx').items()),
```
Comment (eamanu) [inline Lib/test/test_collections.py:1877]: IMO this test is same than above. what about use one of the?

File: Doc/library/collections.rst [hunk 3]
```diff
@@ -272,9 +277,9 @@ For example::
         Return a list of the *n* most common elements and their counts from the
         most common to the least.  If *n* is omitted or ``None``,
         :meth:`most_common` returns *all* elements in the counter.
-        Elements with equal counts are ordered arbitrarily:
+        Elements with equal counts are ordered in the order first encountered:
 
-            >>> Counter('abracadabra').most_common(3)  # doctest: +SKIP
+            >>> Counter('abracadabra').most_common(3)
             [('a', 5), ('r', 2), ('b', 2)]
```
Comment (eamanu) [inline Doc/library/collections.rst]:
```suggestion
             [('a', 5), ('b', 2), ('r', 2)]
```

File: Lib/test/test_collections.py (hunk 2 above)
Comment (rhettinger) [reply Lib/test/test_collections.py:1877]: It's a similar test, but in this case all the inputs are equal.  In the previous test, it was mixed.

File: Doc/library/collections.rst (hunk 3 above)
Comment (rhettinger) [reply Doc/library/collections.rst]: Yes, just found that.

File: Doc/library/collections.rst (hunk 1 above)
Comment (rhettinger) [reply Doc/library/collections.rst]: Yes.

**Pull request merged**